--- /dev/null
+++ ../src/KeycloakGuard.php
@@ -37,6 +37,13 @@
   {
     try {
       $this->decodedToken = Token::decode($this->request->bearerToken(), $this->config['realm_public_key']);
+      if( isset($this->config['clients']) ) {
+        $aKeycloakConfig = json_decode( $this->config['clients'], true );
+        $realmPubKey = $this->config['realm_public_key'];
+        $this->config = $aKeycloakConfig[$this->decodedToken->azp];
+        $this->config['realm_public_key'] = $realmPubKey;
+        //error_log( "Config ". print_r($this->config, true) );
+      }
     } catch (\Exception $e) {
       throw new TokenException($e->getMessage());
     }
