Subject: [PATCH] Renamed vid to tenant on deliver-creds call
---
Index: rest/vendor/robsontenorio/laravel-keycloak-guard/src/KeycloakGuard.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/rest/vendor/robsontenorio/laravel-keycloak-guard/src/KeycloakGuard.php b/rest/vendor/robsontenorio/laravel-keycloak-guard/src/KeycloakGuard.php
--- a/rest/vendor/robsontenorio/laravel-keycloak-guard/src/KeycloakGuard.php	(revision 0ed1611e567f47a2a1a6bf9b491f0b2f143bae32)
+++ b/rest/vendor/robsontenorio/laravel-keycloak-guard/src/KeycloakGuard.php	(date 1684386160652)
@@ -37,6 +37,13 @@
   {
     try {
       $this->decodedToken = Token::decode($this->request->bearerToken(), $this->config['realm_public_key']);
+      if( isset($this->config['clients']) ) {
+        $aKeycloakConfig = json_decode( $this->config['clients'], true );
+        $realmPubKey = $this->config['realm_public_key'];
+        $this->config = $aKeycloakConfig[$this->decodedToken->azp];
+        $this->config['realm_public_key'] = $realmPubKey;
+        //error_log( "Config ". print_r($this->config, true) );
+      }
     } catch (\Exception $e) {
       throw new TokenException($e->getMessage());
     }
